onyo/lib/onyo.py

```diff
@@ -203,155 +203,56 @@ def clear_caches(self,
-        else:
-            # get variables for the begin of the commit message `msg_dummy`
-            dest = None
-            keys_str = ""
-            if keys:
-                keys_str = f" ({','.join(str(x.split('=')[0]) for x in sorted(keys))})"
-            if destination:
-                # Note: This seems to expect relative path in `destination`,
-                #       turns it into absolute and back into relative.
-                #       Double-check where `destination` is used and remove
-                #       resolution:
-                dest = (self.git.root / destination).relative_to(self.git.root)
-            if dest and dest.name == self.ANCHOR_FILE:
-                dest = dest.parent
-
-            # the `msg_dummy` is the way all automatically generated commit
-            # message headers (independently of later adding/shortening of
-            # information) begin, for all commands.
-            msg_dummy = f"{cmd} [{len(changes_to_record)}]{keys_str}"
-            message_subject = self._generate_commit_message_subject(
-                msg_dummy, changes_to_record, dest, max_length)
-
-        return f"{message_subject}"
-
-    @staticmethod
-    def _generate_commit_message_subject(msg_dummy: str,
-                                         changes: list[Path],
-                                         destination: Optional[Path],
-                                         max_length: int = 80) -> str:
-        """Generates a "commit message subject" usable with `git commit`.
-        The function lists the paths of `changes` and will shorten the resulting
-        string to try to include as much user readable information as possible.
+        The function will shorten paths in the resulting string to try to include as much
+        user readable information as possible.
 
         Parameters
         ----------
-        msg_dummy: str
-            The first part of commit message subjects generated by Onyo.
-
-        changes: list of Path
-            The list of paths modified which should be mentioned in the commit
-            message subject.
-
-        destination: Path, optional
-            Destination for assets/directories when the commit contains `mv`s.
+        format_string: str
+            A format string defining the commit message subject to generate.
 
         max_length: int
-            Limit the length of the generated message to `max_length` if
-            possible.
+            An integer specifying the maximal length for generated commit message subjects.
+
+        **kwargs
+            Values to insert into the `format_string`. If values are paths, they will be shortened
+            to include as much user readable information as possible.
 
         Returns
         -------
         str
-            A commit message subject to use with `git commit`.
+            A message suitable as a commit message subject for use with `git commit`.
```

Comment:
Not a blocker for the PR, but this adds nothing:

```suggestion
            A message suitable as a commit message subject.
```